### PR TITLE
Telegram bot beacon

### PR DESCRIPTION
**Pull request by rafaelcaricio**
### What does this PR do?

It is a new beacon to listen messages sent to a Telegram Bot. Example of usage can be seen here: https://github.com/rafaelcaricio/salt-listen-telegram

### What issues does this PR fix or reference?

This PR does not fix any specific issue, but adds more power to SaltStack.

### Tests written?

Yes

### CODE REVIEW

File: salt/beacons/telegram_bot_msg.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    log.debug('telegram_bot_msg beacon starting')
+    ret = []
+    output = {}
+    output['msgs'] = []
+
+    bot = telegram.Bot(config['token'])
+    updates = bot.get_updates(limit=100, timeout=0, network_delay=10)
+
+    log.debug('Num updates: {0}'.format(len(updates)))
+    if len(updates) < 1:
```
Comment (isbm) [inline salt/beacons/telegram_bot_msg.py]: It is always better to say:
```python
if not updates:
...
```

File: salt/beacons/telegram_bot_msg.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            latest_update_id = update.update_id
+
+        if message.chat.username in config['accept_from']:
+            output['msgs'].append(message.to_dict())
+
+    # mark in the server that previous messages are processed
+    bot.get_updates(offset=latest_update_id + 1)
+
+    log.debug('Emitting {0} messages.'.format(len(output['msgs'])))
+    if len(output['msgs']) > 0:
```
Comment (isbm) [inline salt/beacons/telegram_bot_msg.py]: same here:
```python
if output['msgs']:
...
```

**Comment by rafaelcaricio** [comment]
@isbm Thanks for checking it out. Should I send this PR to the upstream repo?

**Comment by isbm** [comment]
@rafaelcaricio If I were you, I would. :smile:

**Pull request closed without merging**